### ISSUE TITLE
BUG - axios-extension getAtoInfo call returns empty ATO configuration

### ISSUE THREAD

**Issue by slin-sap** [SAP/open-ux-tools#515]
### Related Feature
Feature request: _issue number_

### Description

axios-extension getAtoInfo call returns empty ATO configuration

### Steps to Reproduce
1. Add following code in callAFewAbapServices() test function in examples/odata-cli/src/index.ts
```
        const atoSettings = await provider.getAtoInfo();
        console.log(atoSettings);
```
2. Build and run the CLI example project at examples/odata-cli
3. Run command `pnpm start -- abap`

### Expected results

A JS object ATO setting is returned that contains properties such as developmentPrefix, isTransportRequestRequired, transportationMode.  

### Actual results

Empty ATO setting {} is returned.

### Screenshots
If applicable, add screenshots to help explain the problem.

#### Version/Components/Environment	
Add any other context about the problem here
OS:                      
- [ ] Mac OS
- [ ] Windows
- [ ] Other

### Problem
Response has status 200. But the response content is actually application level error message in XML format:

```xml
<?xml version="1.0" encoding="utf-8"?>
<exc:exception xmlns:exc="http://www.sap.com/abapxml/types/communicationframework">
<namespace id="com.sap.adt"/>
<type id="ExceptionResourceNotAcceptable"/>
<message lang="EN">The message content is not acceptable. Accepted content types: application/vnd.sap.adt.ato.settings.v1+xml, application/vnd.sap.adt.ato.settings.v1+json</message>
<localizedMessage lang="EN">The message content is not acceptable. Accepted content types: application/vnd.sap.adt.ato.settings.v1+xml, application/vnd.sap.adt.ato.settings.v1+json</localizedMessage>
<properties/>
</exc:exception>
```

### Fix

The http request implemented in getAtoInfo() in packages/axios-extension/src/abap/abap-service-provider.ts is not accepted by backend. It seems to work fine by adding header:

```javascript
const response = await this.get(ATO_CATALOG_URL_PATH, {
    headers: { Accept: 'application/vnd.sap.adt.ato.settings.v1+xml' }
});
```

### Why was it missed
Specific parameter required by ADT service protocol. Test with actual backend is required.

### How can we avoid this
{if we don’t want to see this type of issues anymore what we should do to prevent}

**Comment by tobiasqueck** [comment]
@slin-sap is this version specific e.g. will it work in all versions or is this something that cannot be used in e.g. older versions?

**Comment by slin-sap** [comment]
@tobiasqueck  @ullasholla 

For #518 ADT support, I made some progress and created a draft PR #519. It gives an example of how I would like to approach ADT client. I used getATOInfo() as an example. I would like to hear your suggestions.

My idea is to add ADT requests as methods in AbapServiceProvider (e.g. getATOInfo).  These methods have a common aspect: "Check if ADT schema is loaded and if not, send discovery request to fetch the schema". I used decorator for this in my PR. Each method for a specific ADT request is decorated with `@adt('ServiceSchemaID')`, and the schema for this service is an optional param that is also decorated `@adtSchema`. 

The @adtSchema parameter decorator is responsible to identify the index of the schema parameter. The @adt decorator is responsible to load ADT schema by calling discovery service, then auto fill the `@adtSchema schema: AdtCollection` parameter by using the parameter index collected from @adtSchema decorator, and finally execute the original getAtoInfo logic. The nice thing is it doesn't matter how many other parameters we have (e.g. dummy params below), the @adt decorator can auto fill the parameter decored with @adtSchema.

```
    // Just notice this 'settings' value is not unique in the schema, will need to add extra parameter to @adt decorator to make 
    // hierarchical lookup to uniquely identify a service schema (AdtCollection). 
    @adt('settings')
    public async getAtoInfo(
        dummy1: string,
        dummy2: string,
        @adtSchema schema?: AdtCollection,
        dummy3?: string
    ): Promise<AtoSettings>
```

The original getAtoInfo implementation accesses the @adtSchema parameter to find out the url path and accept header etc that could be version specific. And provider the correct values to the ServiceProvider.get() http request.
